### ISSUE TITLE
autocomplete -f=godit does not return explicit package name where one is used

### ISSUE THREAD

**Issue by myitcv** [mdempsky/gocode#18]
Consider the following `main.go`:

``` go
package main

import (
    "fmt"
    banana "go/ast"
)

func main() {
    var x *banana.Package
    fmt.Println(x)
    //          ^
}
```

If we invoke `gocode` as follows (101 represents the offset as indicated by the caret above):

```
/path/to/gocode' -sock unix '-in=/my/gopath/src/mydomain.com/g/cmd/banana/main.go' '-f=godit' 'autocomplete' '/my/gopath/src/mydomain.com/g/cmd/banana/main.go' '101'
```

we get the following:

```
1,,1
var x *ast.Package,,x
```

However, as you can see we have explicitly defined the package name for the `"go/ast"` import here... even though the underlying type is indeed `*go/ast.Package`

As a suggestion, I think the reported type should actually be the _package_ qualified type and not the _package name_ qualified type (else the "fix" is more complicated because the output is dependent on the context, explicit package names etc)

Under this proposal the above output would have been:

```
1,,1
var x *go/ast.Package,,x
```

Raising this issue in order to get feedback on this proposal as a first step.

Thanks

**Comment by mdempsky** [comment]
Interesting.  I suppose in general there are three ways to present imported names in hints:
1. Qualified with the package path.
2. Qualified with the package name.
3. Qualified with the import name.  (Usually the same as 2, but sometimes an import is renamed, dot-imported, or the name might not even be directly visible in the current context; for example, var a.X has type b.Y, and we're in a file that has only imported a.)

Currently we're using 2, and you're proposing to switch to 1?  I'm open to that, and it seems like an easy change.  Just worried if the output becomes too noisy.

My sample size of 3 shows other tools aren't consistent here:

godoc and cmd/doc both use package names.  E.g., "go doc net/http/httputil" mentions "http.Request", not "net/http.Request".

guru uses package path.

**Comment by myitcv** [comment]
> Currently we're using 2, and you're proposing to switch to 1? 

Exactly.

> guru uses package path

Probably a good reference point.

Equally option 1 involves a smaller change than option 3. 

If we opt for 3, the change to `gocode` is more involved because context becomes relevant. At the moment (assuming I have understood the code) (godit) [formatting is context free](https://github.com/mdempsky/gocode/blob/e117c263fb7092eedd7cab24f41d90a9d76bd23a/suggest/formatters.go#L50-L55)

**Comment by mdempsky** [comment]
Package path sounds fine to me.

**Comment by myitcv** [comment]
@mdempsky - I've been running with option 1 for a while now. Personally, I don't find the output too noisy. But that is a terrible sample size... and that said, option 3 feels more "natural" (despite the `guru` precedent)

I've just rebased and pushed https://github.com/mdempsky/gocode/pull/19 - build is failing because I've not tried to fix tests as yet.

Thoughts on how to proceed?

**Comment by stamblerre** [comment]
This was fixed by #23.
